ground/src/main/java/com/google/android/ground/persistence/local/LocalDataStore.kt

```diff
@@ -48,76 +43,18 @@ import io.reactivex.Single
  * to allow changes to not rely on prior UI state (i.e., emissions are idempotent).
  */
 interface LocalDataStore {
-  /** Load surveys stored in local database. */
-  val surveys: @Cold Single<ImmutableList<Survey>>
-
-  /** Load last active survey, if any. */
-  fun getSurveyById(id: String): @Cold Maybe<Survey>
-
-  /** Delete stored survey from database. */
-  fun deleteSurvey(survey: Survey): @Cold Completable
-
-  /** Add survey to the database. */
-  fun insertOrUpdateSurvey(survey: Survey): @Cold Completable
-
-  /** Add user to the database. */
-  fun insertOrUpdateUser(user: User): @Cold Completable
-
-  /**
-   * Loads the user with the specified id from the local data store. The returned Single fails with
-   * [java.util.NoSuchElementException] if not found.
-   */
-  fun getUser(id: String): @Cold Single<User>
-
-  /**
-   * Applies the specified [LocationOfInterestMutation] to the local data store, appending the
-   * mutation to the local queue for remote sync.
-   */
-  fun applyAndEnqueue(mutation: LocationOfInterestMutation): @Cold Completable
-
-  /**
-   * Applies the specified [SubmissionMutation] to the local data store, appending the mutation to
-   * the local queue for remote sync.
-   */
-  fun applyAndEnqueue(mutation: SubmissionMutation): @Cold Completable
-
-  /** Applies the specified [SubmissionMutation] to the local data store.. */
-  @Throws(LocalDataStoreException::class) fun apply(mutation: SubmissionMutation): @Cold Completable
-
-  /**
-   * Returns a long-lived stream that emits the full set of LOIs for a survey on subscribe, and
-   * continues to return the full set each time a LOI is added/changed/removed.
-   */
-  fun getLocationsOfInterestOnceAndStream(
-    survey: Survey
-  ): @Cold(terminates = false) Flowable<ImmutableSet<LocationOfInterest>>
-
-  /**
-   * Returns the list of submissions which are not marked for deletion for the specified
-   * locationOfInterest and job.
-   */
-  fun getSubmissions(
-    locationOfInterest: LocationOfInterest,
-    jobId: String
-  ): @Cold Single<ImmutableList<Submission>>
-
-  /** Returns the LOI with the specified UUID from the local data store, if found. */
-  fun getLocationOfInterest(
-    survey: Survey,
-    locationOfInterestId: String
-  ): @Cold Maybe<LocationOfInterest>
-
-  /** Returns the submission with the specified UUID from the local data store, if found. */
-  fun getSubmission(
-    locationOfInterest: LocationOfInterest,
-    submissionId: String
-  ): @Cold Maybe<Submission>
-
-  /**
-   * Returns a long-lived stream that emits the full set of tiles on subscribe and continues to
-   * return the full set each time a tile is added/changed/removed.
-   */
-  val tileSetsOnceAndStream: @Cold(terminates = false) Flowable<ImmutableSet<TileSet>>
+  /** Provides access to [Survey] data in local storage. */
+  var surveyStore: SurveyStore
```

Comment:
If we're splitting up local data stores, for consistency would we then also split up remote ones? If so, should we prefix these with `Local` now to differentiate them conceptually from their eventual remote counterparts?

Reply:
sgtm! Will do